server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -434,6 +437,12 @@ static Settings aggregateIndexSettings(ClusterState currentState, CreateIndexClu
          * that will be used to create this index.
          */
         MetaDataCreateIndexService.checkShardLimit(indexSettings, currentState);
+        if (indexSettings.getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true) == false) {
+            DEPRECATION_LOGGER.deprecatedAndMaybeLog("soft_deletes_disabled",
+                "Index with soft deletes disabled is deprecated and will be rejected. " +
```

Comment:
also adapt message here

Reply:
Adjusted in https://github.com/elastic/elasticsearch/pull/50502/commits/867050d34b0d56783e11f6435afa8d1809bc157c.